include/image-commands.mk

```diff
@@ -142,6 +142,11 @@ define Build/append-dtb
 	cat $(KDIR)/image-$(firstword $(DEVICE_DTS)).dtb >> $@
 endef
 
+define Build/insert-elf-dtb
+	$(TARGET_CROSS)objcopy --update-section \
+				.appended_dtb=$(KDIR)/image-$(firstword $(DEVICE_DTS)).dtb $@
+endef
```

Comment:
This can be a separated commit:
build: image: add insert-elf-dtb helper

Reply:
I do not know if shell uses set -e here, or I should cause make to fail if the dtb insert does not work?
File not found, or section not found return 1.
```
$(shell $(TARGET_CROSS)objcopy --update-section \
				.appended_dtb=$(KDIR)/image-$(firstword $(DEVICE_DTS)).dtb $@);
$(eval $(if $(filter-out 0, $(.SHELLSTATUS)), $(error objcopy failed to insert DTB with return code: $(.SHELLSTATUS))));
```

Reply:
I tried to deliberately corrupt the arguments, and the build did failed:
```
rm -f /home/user/src/openwrt/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/tmp/openwrt-ramips-mt7621-mikrotik_routerboard-750gr3-initramfs-kernel.bin
cp /home/user/src/openwrt/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/vmlinuz-initramfs /home/user/src/openwrt/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/tmp/openwrt-ramips-mt7621-mikrotik_routerboard-750gr3-initramfs-kernel.bin
mipsel-openwrt-linux-musl-objcopy --update-section .appended_dtb=/home/user/src/openwrt/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/image-mt7621_mikrotik_routerboard-750gr3.dtb.blablabla /home/user/src/openwrt/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/tmp/openwrt-ramips-mt7621-mikrotik_routerboard-750gr3-initramfs-kernel.bin
mipsel-openwrt-linux-musl-objcopy: cannot open: /home/user/src/openwrt/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/image-mt7621_mikrotik_routerboard-750gr3.dtb.blablabla: No such file or directory
make[4]: *** [Makefile:230: /home/user/src/openwrt/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/tmp/openwrt-ramips-mt7621-mikrotik_routerboard-750gr3-initramfs-kernel.bin] Error 1
make[4]: Leaving directory '/home/user/src/openwrt/target/linux/ramips/image'
make[3]: *** [Makefile:25: install] Error 2
make[3]: Leaving directory '/home/user/src/openwrt/target/linux/ramips'
make[2]: *** [Makefile:13: install] Error 2
```